test/hazelcast_json_value/HazelcastJsonValueTest.js

```diff
@@ -0,0 +1,199 @@
+                expect(value).to.be.an.instanceof(HazelcastJsonValue);
+                expect(value).to.be.deep.equal(hzJsonValue);
+                expect(value.parse()).to.deep.equal(object);
+                return map.destroy();
+            }).then(function () {
+                return client.shutdown();
+            });
+        });
+    });
+
+    it('Storing invalid Json strings with HazelcastJsonValueSerializer', function () {
+        var invalidString = '{a}';
+        var hzJsonValueInvalid = new HazelcastJsonValue(invalidString);
+        var config = new Config.ClientConfig();
+        config.serializationConfig.jsonDeserializationType = Config.JsonDeserializationType.HAZELCAST_JSON_VALUE;
+        return HazelcastClient.newHazelcastClient(config).then(function (client) {
+            return client.getMap('jsonTest').then(function (mp) {
+                map = mp;
+                return map.put(1, hzJsonValueInvalid);
+            }).then(function () {
+                return map.get(1);
+            }).then(function (value) {
+                expect(value).to.be.an.instanceof(HazelcastJsonValue);
+                expect(value).to.be.deep.equal(hzJsonValueInvalid);
+                expect(value.parse.bind(value)).to.throw(SyntaxError);
+                return map.destroy();
+            }).then(function () {
+                return client.shutdown();
+            });
+        });
+    });
+
+    it('Querying over JavaScript objects', function () {
+        var objects = [
+            [0, {'a': 1}],
+            [1, {'a': 3}]
+        ];
+        return HazelcastClient.newHazelcastClient().then(function (client) {
+            return client.getMap('jsonTest').then(function (mp) {
+                map = mp;
+                return map.putAll(objects);
+            }).then(function () {
+                return map.valuesWithPredicate(Predicates.greaterThan('a', 2));
+            }).then(function (values) {
+                expect(values.toArray()).to.deep.equal([objects[1][1]]);
+                return map.destroy();
+            }).then(function () {
+                return client.shutdown();
+            });
+        });
+    });
+
+    it('Querying over nested attributes', function () {
+        var objects = [
+            [0, {'a': 1, 'b': {'c': 1}}],
+            [1, {'a': 3, 'b': {'c': 3}}]
+        ];
+        return HazelcastClient.newHazelcastClient().then(function (client) {
+            return client.getMap('jsonTest').then(function (mp) {
+                map = mp;
+                return map.putAll(objects);
+            }).then(function () {
+                return map.valuesWithPredicate(Predicates.greaterThan('b.c', 2));
+            }).then(function (values) {
+                expect(values.toArray()).to.deep.equal([objects[1][1]]);
+                return map.destroy();
+            }).then(function () {
+                return client.shutdown();
+            });
+        });
+    });
+});
```

Comment:
Can you expand the tests a bit more with cases like putting Object and HazelcastJsonValue mixed in the same map, querying over `__key` instead of values. Putting json values as keys etc.